### ISSUE TITLE
NoMemoryError (failed to allocate memory)

### ISSUE THREAD

**Issue by Brunomm** [msgpack/msgpack-ruby#180]
When I use the following string, a memory error occurs.
```ruby
x = "\x82\xA9condition\xA2OR\xA6groups\x92\x82\xA9condition\xA2or\xA7filters\x91\x83\xA3col\xAAbirth_date\x
A3ope\xA5month\xA3val\xA12\x82\xA9condition\xA2or\xA7filters\x91\x83\xA3col\xADshare_capital\xA3ope\xB3greater_or_equal_
th\xDD\x98[)\x8DL\f\v\x8C\f"

MessagePack.unpack x

# OR

b = "gqljb25kaXRpb26iT1KmZ3JvdXBzkoKpY29uZGl0aW9uom9yp2ZpbHRlcnORg6Njb2yqYmlydGhfZGF0ZaNvcGWlbW9udGijdmFsoTKCqWNvbmRp
dGlvbqJvcqdmaWx0ZXJzkYOjY29srXNoYXJlX2NhcGl0YWyjb3BlsGdyZWF0ZXJvcmVxdWFsdG-jdmFspjUwMC4wMA"

MessagePack.unpack Base64.decode64(b)
```

I'm using [this tool](http://output.jsbin.com/cayuhox) to test and generate a Base64 Pack where I later decode the base64 and try to run the unpack.
When using the following JSON the error is occurring:
```json
{"condition":"OR","groups":[{"condition":"or","filters":[{"col":"birth_date","ope":"month","val":"2"}]},{"condition":"or","filters":[{"col":"share_capital","ope":"greaterorequalto","val":"500.00"}]}]}
```

But it's curious when I substitute the word birth_date for birthdate the problem doesn't happen:
```json
{"condition":"OR","groups":[{"condition":"or","filters":[{"col":"birthdate","ope":"month","val":"2"}]},{"condition":"or","filters":[{"col":"share_capital","ope":"greaterorequalto","val":"500.00"}]}]}
```

My ruby version is 2.7.0.

![image](https://user-images.githubusercontent.com/2380733/75083891-15f31c80-54fb-11ea-9507-93d5250b7be5.png)

**Comment by tagomoris** [comment]
I tried all of your example code, but it works well without NoMemoryError. The first two unpack example seems having invalid input data and those failed with EOFError (because of insufficient valid input).
(I tried on macOS with Ruby 2.7.0.)

@Brunomm Could you add your environment in detail (OS, output of `ruby --version`) and also share the exact script you tried (using gist or something)?

**Comment by Brunomm** [comment]
Hi @tagomoris 

My version of Ruby and OS is `ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]`.

I believe that I have found a solution for my specific case.

Here is the gist with an example where the error occurs and with a solution I found.
https://gist.github.com/Brunomm/ba9de5fcf4e007b46561d2cfbeffaeeb

As I am using a [lib in Javascript](https://www.npmjs.com/package/json-url) to make the pack, I realized that it converts base64 to be compatible to send via urls.
With that, I found that if I use `Base64.urlsafe_decode64` the problem does not occur.
